com.unity.multiplayer.mlapi/Tests/Runtime/Metrics/NetworkObjectMetricsTests.cs

```diff
@@ -161,28 +163,29 @@ public IEnumerator TrackMultipleNetworkObjectDestroySentMetric()
             anotherNetworkObject.NetworkManagerOwner = m_Server;
             anotherNetworkObject.Spawn();
 
+            yield return new WaitForSeconds(0.2f);
+
             var waitForMetricEvent = new WaitForMetricValues<ObjectDestroyedEvent>(m_ServerMetrics.Dispatcher, MetricNames.ObjectDestroyedSent);
-            yield return waitForMetricEvent.WaitForAFewFrames();
 
             NetworkObject.NetworkHide(new List<NetworkObject>{m_NewNetworkObject, anotherNetworkObject}, m_Client.LocalClientId);
 
-            yield return waitForMetricEvent.Wait(60);
+            yield return waitForMetricEvent.WaitForMetricsReceived();
 
-            var objectDestroyedSentMetricValues = waitForMetricEvent.EnsureMetricValuesHaveBeenFound();
+            var objectDestroyedSentMetricValues = waitForMetricEvent.AssertMetricValuesHaveBeenFound();
             // As there's a client and server, this event is emitted twice.
             Assert.AreEqual(2, objectDestroyedSentMetricValues.Count);
-
-            var networkIDFound = true;
-            var networkNameFound = true;
-            // not sure that we can guarantee the order of these so just ensure the data is in the received metrics.
-            foreach (var metricValue in objectDestroyedSentMetricValues)
-            {
-                Assert.AreEqual(m_Client.LocalClientId, metricValue.Connection.Id);
-                networkIDFound &= metricValue.NetworkId.NetworkId == m_NewNetworkObject.NetworkObjectId || metricValue.NetworkId.NetworkId == anotherNetworkObject.NetworkObjectId;
-                networkNameFound &= metricValue.NetworkId.Name == m_NewNetworkObject.name || metricValue.NetworkId.Name == anotherNetworkObject.name;
-            }
-            Assert.IsTrue(networkIDFound);
-            Assert.IsTrue(networkNameFound);
+            Assert.That(
```

Comment:
@josiemessa If you get a chance to see this, just to let you know that NUnit has a cool feature for this sort of check!